llvm/utils/TableGen/SearchableTableEmitter.cpp

```diff
@@ -426,16 +428,25 @@ void SearchableTableEmitter::emitLookupFunction(const GenericTable &Table,
     OS << "    return nullptr;\n\n";
   }
 
-  OS << "  struct KeyType {\n";
-  for (const auto &Field : Index.Fields) {
-    OS << "    " << searchableFieldType(Table, Index, Field, TypeInTempStruct)
-       << " " << Field.Name << ";\n";
+  bool ShouldReturnRange = Index.ReturnRange;
+  if (ShouldReturnRange)
```

Comment:
Is `ShouldReturnRange` only compatible with `IsPrimary==true`?

Reply:
IIUC your comment - `ShouldReturnRange` can be `true` irrespective of `IsPrimary` is true or not. 
`ShouldReturnRange` is populated by the `ReturnRange` field of `SearchIndex`. So if `ReturnRange` is set to true for a non-primary function, then also `ShouldReturnRange` will hold true. An example to illustrate the same is already added in `ReturnRange.td` file.